sympy/physics/mechanics/joint.py

```diff
@@ -1707,3 +1707,255 @@ def _set_linear_velocity(self):
             self.planar_speeds[1] * self.planar_vectors[1])
         self.child.masscenter.v2pt_theory(self.child_point, self.parent.frame,
                                           self.child.frame)
+
+
+class SphericalJoint(Joint):
+    """Spherical (Ball-and-Socket) Joint.
+
+    .. image:: SphericalJoint.svg
+        :align: center
+        :width: 600
+
+    Explanation
+    ===========
+
+    A spherical joint is defined such that the child body is free to rotate in
+    any direction, without allowing a translation of the ``child_point``. As can
+    also be seen in the image the ``parent_point`` and ``child_point`` are fixed
+    on top of each other, i.e. the joint_point. This rotation is defined using
+    :func:`parent_interframe.orient(child_interframe, rot_type, amounts,
+    rot_order)<sympy.physics.vector.frame.ReferenceFrame.orient>` method. The
+    default rotation consists of three relative rotations, i.e. body fixed
+    rotations. Based on the direction cosine matrix following from these
+    rotations the angular velocity is computed based on the generalized
+    coordinates and generalized speeds.
```

Comment:
Why do some references to attributes get double backticks (e.g. ` ``parent_point`` `) while others don't (e.g. `joint_point`)?
```suggestion
    A spherical joint is defined such that the child body is free to rotate in
    any direction, without allowing a translation of the ``child_point``. As can
    also be seen in the image, the ``parent_point`` and ``child_point`` are fixed
    on top of each other, i.e. the joint_point. This rotation is defined using the
    :func:`parent_interframe.orient(child_interframe, rot_type, amounts,
    rot_order)<sympy.physics.vector.frame.ReferenceFrame.orient>` method. The
    default rotation consists of three relative rotations, i.e. body fixed
    rotations. Based on the direction cosine matrix following from these
    rotations, the angular velocity is computed based on the generalized
    coordinates and generalized speeds.
```

Reply:
Forgot that one